src/mainview.cpp

```diff
@@ -1,21 +1,38 @@
 #include "mainview.h"
 
+#include "FilesystemSearchProvider.h"
 #include "config.h"
 
 #include <QFile>
 #include <QJsonArray>
 #include <QJsonDocument>
 #include <QJsonObject>
+#include <QScrollArea>
 #include <QMessageBox>
 
 using namespace ui;
 
 CMainView::CMainView( QWidget *pParent ) :
 	QDialog( pParent )
 {
-	auto pLayout = new QVBoxLayout( this );
+	CFileSystemSearchProvider provider;
+	char *installDir = new char[1048];
+	provider.GetAppInstallDir( qgetenv("SteamAppId").toInt(), installDir, 1048 );
+	m_pInstallDir = QString( installDir );
+	delete[] installDir;
+
+
+	auto pLayout = new QGridLayout( this );
```

Comment:
Also formatting in this general area too

Reply:
> I know I made a habit of this, and it's an awful habit, but could we change the autos to what they're meant to be?

CLion tells me that auto is correct, I also feel like auto is fine.

Reply:
I mean auto is equivalent to `var` in C#, it of course will be correct, but I'd like it if the correct types were used.

Reply:
The reason for all of this is:
> Clang-Tidy: Use auto when initializing with new to avoid duplicating the type name

Reply:
Also something else unrelated to that, pulled it myself and I can't compile the sdk launcher since I gave errors in sapp since I forgot to add `set(CMAKE_CXX_STANDARD 20)` to the cmake file, could you probably do that in this pr? Saves you from having to rebase again.

Reply:
IIRC it's 20 for Linux and 23 for Windows, correct?

Reply:
20 works fine when I tested it jusst

Reply:
IIRC, it was either ender or someone else who explained to me that Windows requires 23 for modern C++ modules.

Reply:
Added 20/23 support in
[424692741121745c1de69a92c3f6e3355391346d](https://github.com/AWildErin/sdklauncher/pull/9/commits/424692741121745c1de69a92c3f6e3355391346d)